### PR TITLE
[CI] Add link-checking workflow

### LINKED ISSUE #667: Provide CI check to verify links
There is some script code there to check links, but it doesn't work.

### PR DESCRIPTION
**Pull request by nate-double-u**
Adding a github workflow to run the link checker on PR.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Fixes #667

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,6 +2,8 @@
   "scripts": {
     "_build": "hugo --cleanDestinationDir -e dev -DFE",
     "_check-links": "make check-links",
+    "_diff:check": "git diff --name-only --exit-code",
+    "_diff:fail": "git diff --name-only --exit-code || exit 1",
```
Comment (chalin) [inline package.json:6]:
```suggestion
```

File: package.json
Comment (chalin) [inline package.json:1]: In the `postbuild:*` scripts, replace `_check-links` by `_check-links--warn` as is done in https://github.com/chalin/docsy-starter/blob/main/package.json. Also get that new script from there.

By making that change, it'll fix the build failure, and delegate the failure reporting to the GH workflow.

**Review by chalin — COMMENTED** [review]
You're close, but there are a few issues that we can tackle incrementally. For a first iteration of fixes, see the inline comments.

**Comment by nate-double-u** [comment]
Updated, now failing condition looks to be working:
Screenshot 2024-10-18 at 1 23 23 PM

Now removing edit to test failure condition, next push should build correctly.

**Comment by nate-double-u** [comment]
Build working, no broken links found, github action doing the reporting in both passing and failing scenarios.

Screenshot 2024-10-18 at 1 27 48 PM

Next push rebases this branch to bring this branch up to date with main.

**Comment by nate-double-u** [comment]
This is ready for review now. Thanks for the help @chalin!

**Review by JorTurFer — APPROVED** [review]

**Comment by JorTurFer** [comment]
nice job! Thanks 🙇

**Pull request merged**